cmd/cluster/core/create.go

```diff
@@ -265,10 +266,25 @@ func GetAPIServerAddressByNode(ctx context.Context) (string, error) {
 func Validate(ctx context.Context, opts *CreateOptions) error {
 	if !opts.Render {
 		client := util.GetClientOrDie()
+		// Validate HostedCluster CustomResource is installed
+		// If the CustomResource doesn't exist, it might indicated `hypershift install` wasn't called
+		hcCRD := &apiextensionsv1.CustomResourceDefinition{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "hostedclusters.hypershift.openshift.io",
+			},
+		}
+		if err := client.Get(ctx, crclient.ObjectKeyFromObject(hcCRD), hcCRD); err != nil {
```

Comment:
Kind of a nit I guess, but is it possible the user has RBAC permissions to manipulate hostedclusters but not CRDs? I'm not sure this logic has anything to do with validating hostedclusters (as the fn name and context implies).

What server error is returned when you GET a resource with an unknown type? Could you just interpret that in the response for the get if you want to try and provide more context about the failure?

Reply:
You right about the permissions issue, I will remove this part

Reply:
Done
I will check about if can get more info from the error, will do this in another PR